_posts/2020-05-02-schedule-your-functions.md

```diff
@@ -0,0 +1,264 @@
+
+```bash
+faas-cli new lockbot --lang python3-flask-debian --prefix <dockerhub_username> && mv lockbot.yml stack.yml
+```
+
+### Generate access token
+
+Go to the [personal access tokens](https://github.com/settings/tokens) page in GitHub and press the `Generate new token` button. The bot will use this to authenticate to the repository you chose.
+
+Now copy the generated token and create `auth-token` secret using the `faas-cli` where you need to replace the `<token>` with the actual copied access token:
+
+```bash
+faas-cli secret create auth-token --from-literal='<token>'
+```
+
+### Configure the function
+
+Open the function's configuration file called `stack.yml`. Append the following lines to the file:
+
+```yml
+    environment:
+      github_username: <github_username>
+      github_repository: <repository_to_watch>
+    secrets:
+      - auth-token
+    annotations:
+      topic: cron-function
+      schedule: "0 0 * * *"
+```
+
+Replace `<repository_to_watch>` with one of your own repositories and `<github_username>` with your GitHub username, which should also own the repository. The whole `stack.yml` file should look something like this:
+
+```yml
+version: 1.0
+provider:
+  name: openfaas
+  gateway: <gateway_url>
+functions:
+  lockbot:
+    lang: python3-flask-debian
+    handler: ./lockbot
+    image: <your_dockerhub_username>/lockbot:latest
+    environment:
+      github_username: <github_username>
+      github_repository: <repository_to_watch>
+      days_before_outdated: 90
+    secrets:
+      - auth-token
+    annotations:
+      topic: cron-function
+      schedule: "0 0 * * *"
+```
+
+### Augment the code
+
+Inside the `lockbot` folder you can see the `handler.py` file which contains the `handle` method which is the entry point to your function. Replace what's inside with the lockbot's code:
+
+```python
+import os
+import requests
+from datetime import datetime
+
+
+def get_difference_in_days(string_date):
+    # GitHub date format: 2019-01-13T21:17:09Z
+    created_date = string_date.split('T')[0].split('-')
+    for i in range(0, len(created_date)):
+        created_date[i] = int(created_date[i])
+    created_date = datetime(created_date[0], created_date[1], created_date[2])
+    return datetime.now() - created_date
+
+
+def lock_issue(issue_number, auth_header, username, repo):
+    r = requests.put(
+        url=f"https://api.github.com/repos/{username}/{repo}/issues/{issue_number}/lock", headers=auth_header)
+    if r.status_code == 204:
+        return f"Issue: {issue_number} locked"
+
+
+def get_authorization_header():
+    with open("/var/openfaas/secrets/auth-token") as file:
+        return {'Authorization': f"token {file.read().strip()}"}
+
+
+def handle(req):
+    username = os.getenv("github_username")
+    repo = os.getenv("github_repository")
+    days_before_outdated = int(os.getenv("days_before_outdated"))
+    r = requests.get(f"https://api.github.com/repos/{username}/{repo}/issues")
```

Comment:
Again I think this should be using the Python SDK

Reply:
Will switch to that as above :+1:

Reply:
PyGithub in action now :heavy_check_mark: